tests/Unit/Models/CourseTest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Tests\Unit\Models;
+
+use Tests\TestCase;
+use App\Models\Course;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+
+class CourseTest extends TestCase
+{
+    /**
+     * A basic unit test example.
+     *
+     * @return void
+     */
+    public function test_groups_relation()
+    {
+        $course = new Course();
+        $relation = $course->groups();
+        $this->assertInstanceOf(HasMany::class, $relation);
+    }
+
+    public function test_user_relation()
+    {
+        $course = new Course();
+        $relation = $course->user();
+        $this->assertInstanceOf(BelongsTo::class, $relation);
+    }
+
+    public function test_users_relation()
+    {
+        $course = new Course();
+        $relation = $course->users();
+        $this->assertInstanceOf(BelongsToMany::class, $relation);
+        $this->assertEquals('course_id', $relation->getForeignPivotKeyName());
+        $this->assertEquals('user_id', $relation->getRelatedPivotKeyName ());
```

Comment:
giải thích các function test

Reply:
để e thêm cmt

Reply:
+ **test_groups_relation()** : kiểm tra quan hệ hasMany của course với group( 1 lớp học có nhiều nhóm)
   -  $this->**assertInstanceOf(HasMany::class, $relation)** : kiểm tra $relation có phải là 1 quan hệ hasMany hay không
+ **test_user_relation()** :  kiểm tra quan hệ belongsTo của course với user( 1 lớp học được dạy bởi 1 giảng viên)
   -  $this->**assertInstanceOf(BelongsTo::class, $relation)** :  kiểm tra  $relation có phải là 1 quan hệ belongsTo hay 
      không
+ **test_users_relation()** : kiểm tra quan hệ belongsToMany của course với user( 1 lớp chứa nhiều học sinh)
   -  $this->**assertInstanceOf(BelongsToMany::class, $relation)** : kiểm tra  $relation có phải là 1 quan hệ 
      belongsToMany  hay không
   -  $this->**assertEquals('course_id', $relation->getForeignPivotKeyName())** : kiểm tra khóa ngoại trong bảng pivot 
      tham chiếu tới bảng courses
   -  $this->**assertEquals('user_id', $relation->getRelatedPivotKeyName ())** : kiểm tra khóa liên quan trong bảng 
      pivot( là khóa ngoại tham chiếu tới bảng còn lại- bảng users)